pkg/ansible/runner/runner_test.go

```diff
@@ -177,45 +142,63 @@ func TestNew(t *testing.T) {
 			}
 
-			if testRunnerStruct.maxRunnerArtifacts != tc.watch.MaxRunnerArtifacts {
-				t.Fatalf("Unexpected maxRunnerArtifacts %v expected maxRunnerArtifacts %v", testRunnerStruct.maxRunnerArtifacts, tc.watch.MaxRunnerArtifacts)
+			if testRunnerStruct.maxRunnerArtifacts != testWatch.MaxRunnerArtifacts {
+				t.Fatalf("Unexpected maxRunnerArtifacts %v expected maxRunnerArtifacts %v", testRunnerStruct.maxRunnerArtifacts, testWatch.MaxRunnerArtifacts)
 			}
 
 			// Check the cmdFunc
-			checkCmdFunc(t, testRunnerStruct.cmdFunc, tc.watch.Role, tc.watch.Playbook, tc.watch.MaxRunnerArtifacts, testRunnerStruct.maxRunnerArtifacts)
+			checkCmdFunc(t, testRunnerStruct.cmdFunc, testWatch.Playbook, testWatch.Role, testWatch.AnsibleVerbosity)
 
 			// Check finalizer
-			if testRunnerStruct.Finalizer != tc.watch.Finalizer {
-				t.Fatalf("Unexpected finalizer %v expected finalizer %v", testRunnerStruct.Finalizer, tc.watch.Finalizer)
+			if testRunnerStruct.Finalizer != testWatch.Finalizer {
+				t.Fatalf("Unexpected finalizer %v expected finalizer %v", testRunnerStruct.Finalizer, testWatch.Finalizer)
 			}
 
-			if tc.watch.Finalizer != nil {
-				if testRunnerStruct.Finalizer.Name != tc.watch.Finalizer.Name {
-					t.Fatalf("Unexpected finalizer name %v expected finalizer name %v", testRunnerStruct.Finalizer.Name, tc.watch.Finalizer.Name)
+			if testWatch.Finalizer != nil {
+				if testRunnerStruct.Finalizer.Name != testWatch.Finalizer.Name {
+					t.Fatalf("Unexpected finalizer name %v expected finalizer name %v", testRunnerStruct.Finalizer.Name, testWatch.Finalizer.Name)
 				}
 
-				if len(tc.watch.Finalizer.Vars) == 0 {
-					checkCmdFunc(t, testRunnerStruct.finalizerCmdFunc, tc.watch.Finalizer.Role, tc.watch.Finalizer.Playbook, tc.watch.MaxRunnerArtifacts, testRunnerStruct.maxRunnerArtifacts)
+				if len(testWatch.Finalizer.Vars) == 0 {
+					checkCmdFunc(t, testRunnerStruct.cmdFunc, testWatch.Finalizer.Playbook, testWatch.Finalizer.Role, testWatch.AnsibleVerbosity)
 				} else {
 					// when finalizer vars is set the finalizerCmdFunc should be the same as the cmdFunc
-					checkCmdFunc(t, testRunnerStruct.finalizerCmdFunc, tc.watch.Role, tc.watch.Playbook, tc.watch.MaxRunnerArtifacts, testRunnerStruct.maxRunnerArtifacts)
+					checkCmdFunc(t, testRunnerStruct.finalizerCmdFunc, testWatch.Playbook, testWatch.Role, testWatch.AnsibleVerbosity)
 				}
 			}
-
 		})
 	}
 }
+
+func TestAnsibleVerbosityString(t *testing.T) {
+	testCases := []struct {
+		verbosity      int
+		expectedString string
+	}{
+		{verbosity: -1, expectedString: ""},
+		{verbosity: 0, expectedString: ""},
+		{verbosity: 1, expectedString: "-v"},
+		{verbosity: 2, expectedString: "-vv"},
+	}
+
+	for _, tc := range testCases {
+		gotString := ansibleVerbosityString(tc.verbosity)
+		if tc.expectedString != gotString {
+			t.Fatalf("Unexpected string %v expected %v", gotString, tc.expectedString)
```

Comment:
Should not possible to add a test to ensure that if the -vv, for example, be used then the logs are with this verbose? Could we not check that the output which jut will occur in this scenario was not printed?

Reply:
I think it would be possible to add a functional test to do this, yes. I do not believe it would be easy to construct a test in `runner_test.go` that does as you describe.

Reply:
Wdyt about using the e2e tests for it?